tests/End2EndTestCase.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Delivery;
+
+use Contentful\Delivery\Client;
+use Contentful\Log\ArrayLogger;
+
+class End2EndTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string
+     */
+    protected static $cacheDir = '';
+
+    public static function setUpBeforeClass()
+    {
+        self::$cacheDir = __DIR__ . '/../build/cache';
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return Client
+     */
+    protected function getClient($key)
+    {
+        $options = getenv('CONTENTFUL_CDA_SDK_TESTING_URL')
+            ? ['uriOverride' => getenv('CONTENTFUL_CDA_SDK_TESTING_URL')]
+            : [];
+
+        switch ($key) {
```

Comment:
Mostly because not all methods would have meaningful names.
For instance `cfexampleapi` and `88dyiqcr7go8` (and their respective `_preview` versions) are for all intents and purposes the same, but the things they need for testing are in two different spaces.

Some of them would be hard to describe using a name, so we might as well not do that and just use a key...